### DISCUSSION TITLE
New to Open-Source Contribution

### DISCUSSION THREAD

**Discussion by viralcodex** [bigskysoftware/htmx#2492] [General]
Hello all, myself Aviral from India and I'm currently a fresher with 1 year as an SDE just out of college.

I was willing to start contributing to open source projects and learn and was looking for some projects I can work on apart from my regular work. HTMX caught my eye as an interesting project from 'Primeagen' youtube video I was recently watching.

I want to get some insights here as to how to proceed with starting here, any small issues I can go through (solved/unsolved) to understand the way of working/solving issues, or any insights on how to become better as a contributer.

I'm already going to through the codebase and discussions to understand more.

Would love to hear from a great community and really excited to get my first contribution here :)

**Comment by nickchomey** [comment]
I think you will find that in most repos people are not going to be willing to guide you through becoming a contributor/developer. It simply isn't worth their time.

One of the most important things in open source, development in general, and especially in life in general, is that you are a self-starter and can find and fix your problems without guidance from others. 

So, I suggest that you first start with finding a project that you want to build yourself - it could be anything. And then explore for tools/libraries/repos that might be useful for it and try to integrate them into your project. When you find something that doesn't work as you desire, then try to fix it and submit a Pull Request for review.

You can also just search the existing issues in the repos of libraries that you're using and see if there's one you have interest in and able to solve, and then do it.

I hope this helps.

**Reply by viralcodex** [reply]
HI thanks for the reply and yeah I understand that and I think that what I wrote sounds different than what I wanted, just needed a little direction where to start in this repo as I'm using HTMX for some of my personal projects (simple ones) and just wanted to know if I should have a look through issues first or through the codebase and I've actually understood some of the codebase. I just need to know where to start as I have seen the implementation of htmx, not constant guidance actually, so that I can build my path in here and be able to contribute in a few months (ofc I know I cannot start contributing in a day or a week).

Thanks for replying

**Reply by nickchomey** [reply]
Again, just use htmx (or any library) in your own project and if you find a bug, then fix it and submit a PR. Or review the existing issues and see if there is one that you can help with. 

The codebase is pretty simple - it is one file and can easily be step debugged in the browser - so you shouldn't really need any help getting oriented. 

If you find that there's a feature missing that you would like, then create an issue proposing its addition and the maintainers can approve or reject it, or suggest it as a plugin. They will perhaps then point you in the right direction if there is something that isn't clear to you. 

Every library repo has different contributor guidelines, but what I've just described should apply to most of them.
